test/set.js

```diff
@@ -25,9 +25,10 @@ assert.equal(result.stderr, 'ls: no such file or directory: file_doesnt_exist\n'
 
 // set -e
 var result = shell.exec('node -e \"require(\'../global\'); set(\'-e\'); ls(\'file_doesnt_exist\'); echo(1234);\"');
-assert.equal(result.code, 1);
+var uncaughtErrorExitCode = process.version.indexOf('v0.10') === 0 ? 8 : 1;
```

Comment:
You're right. Not sure exactly when that behavior changed. I'll try to figure that out and make the test work everywhere.